### PR TITLE
feat(cli) - allow override for typeConfiguration.json file

### LINKED ISSUE #956: Add option to specify the typeconfiguration.json file to use
Testing / submitting hooks/resources requires copying over specific typeconfiguration.json files to ~/.cfn-cli/typeConfiguration.json

It would be useful to either default to the local resource/hook directory where the cfn commands are being run and/or be able to specify which typeConfiguration.json file to use.

### PR DESCRIPTION
**Pull request by mmaeng**
Closes #956 

new --typeconfig option to override default typeConfiguration.json file along with unittests and addition to README file.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -60,7 +60,7 @@ cfn test --log-group-name cw_log_group --log-role-arn log_delivery_role_arn # Ha
 ```
 
 Note:
-* To use your type configuration in contract tests, you will need to save your type configuration json file in `~/.cfn-cli/typeConfiguration.json`.
+* To use your type configuration in contract tests, you will need to save your type configuration json file in `~/.cfn-cli/typeConfiguration.json` or specify the file with `--typeconfig <alternate file including location>`
```
Comment (ericzbeard) [inline README.md:63]: We should be more specific about the format of the argument. Is it an absolute path?

**Review by kddejong — APPROVED** [review]

File: src/rpdk/core/contract/type_configuration.py [hunk 2]
```diff
@@ -6,48 +6,50 @@
...
 class TypeConfiguration:
     TYPE_CONFIGURATION = None
 
     @staticmethod
-    def get_type_configuration():
+    def get_type_configuration(typeconfigloc):
+        if typeconfigloc:
+            type_config_file_path = typeconfigloc
+        else:
+            type_config_file_path = "~/.cfn-cli/typeConfiguration.json"
```
Comment (ericzbeard) [inline src/rpdk/core/contract/type_configuration.py:18]: I think `~` fails in some scenarios. We should use the standard function for getting the user's home directory on all platforms.

File: src/rpdk/core/contract/type_configuration.py [hunk 3]
```diff
@@ -6,48 +6,50 @@
...
         LOG.debug(
-            "Loading type configuration setting file at '~/.cfn-cli/typeConfiguration.json'"
+            "Loading type configuration setting file at %s",
+            type_config_file_path,
         )
         if TypeConfiguration.TYPE_CONFIGURATION is None:
             try:
                 with open(
-                    os.path.expanduser(TYPE_CONFIGURATION_FILE_PATH), encoding="utf-8"
+                    os.path.expanduser(type_config_file_path), encoding="utf-8"
```
Comment (ericzbeard) [inline src/rpdk/core/contract/type_configuration.py:27]: Maybe this addresses my previous comment.

**Review by ericzbeard — CHANGES_REQUESTED** [review]

File: src/rpdk/core/contract/type_configuration.py (hunk 3 above)
Comment (mmaeng) [reply src/rpdk/core/contract/type_configuration.py:27]: That is correct.  The path.expanduser function works on posix and windows.

File: src/rpdk/core/contract/type_configuration.py (hunk 2 above)
Comment (mmaeng) [reply src/rpdk/core/contract/type_configuration.py:18]: this is expanded by python's os.path.expanduser function and works across posix and windows.

File: README.md (hunk 1 above)
Comment (mmaeng) [reply README.md:63]: Let me add in more details

**Review by ericzbeard — APPROVED** [review]

**Pull request merged**